### PR TITLE
Update Logtalk support to the latest language spec

### PR DESCRIPTION
**Pull request by pmoura**
*Description of changes:*

- Add syntax highlight support for the `threaded_cancel/1` and `uninstantiation_error/1` built-in methods
- Add escape sequences support for double-quoted terms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/mode/logtalk_highlight_rules.js [hunk 1]
```diff
@@ -86,7 +74,7 @@ var LogtalkHighlightRules = function() {
          { token: 'support.function.control.logtalk',
            regex: '\\b(?:true|fa(?:il|lse)|repeat|(?:instantiation|system)_error)\\b(?![-!(^~])' },
          { token: 'support.function.control.logtalk',
-           regex: '\\b((?:type|domain|existence|permission|representation|evaluation|resource|syntax)_error)(?=[(])' },
+           regex: '\\b((?:uninstantiation|type|domain|existence|permission|representation|evaluation|resource|syntax)_error)(?=[(])' },
```
Comment (oykuyilmaz) [inline src/mode/logtalk_highlight_rules.js:77]: do we need here `instantiation_error` as well? [Ref](https://logtalk.org/manuals/refman/methods/error_handling_methods.html).

Comment (pmoura) [reply src/mode/logtalk_highlight_rules.js:77]: No. `instantiation_error/0` is above that line as it doesn't have any arguments unlike e.g. `uninstantiation_error/1` that have one argument (and thus `uninstantiation_error` must be followed by a `(` character; no spaces between allowed).

File: src/mode/logtalk_highlight_rules.js [hunk 2]
```diff
@@ -170,7 +158,8 @@ var LogtalkHighlightRules = function() {
          { token: 'punctuation.definition.string.begin.logtalk',
            regex: '"',
            push: 
-            [ { token: 'constant.character.escape.logtalk', regex: '\\\\.' },
+            [ { token: 'constant.character.escape.logtalk',
+                regex: '\\\\([\\\\abfnrtv"\']|(x[a-fA-F0-9]+|[0-7]+)\\\\)' },
```
Comment (oykuyilmaz) [inline src/mode/logtalk_highlight_rules.js:162]: should we consider other escape sequences like `\UXXXXXXXX` or `\uXXXX`?

Comment (pmoura) [reply src/mode/logtalk_highlight_rules.js:162]: Those escape sequences are implemented only by some of the backend Prolog compilers supported by Logtalk, notably only for those that support Unicode. They are slowly becoming a de facto standard feature. But they are also rarely used.

**Review by oykuyilmaz — APPROVED** [review]

**Pull request merged**